modules/sync/admin-sync.php

```diff
@@ -230,4 +250,35 @@ public function __call( $func, $args ) {
 			E_USER_ERROR
 		);
 	}
+
+	/**
+	 * Checks the context, and returns the value of `from_post_id` and `new_lang`.
+	 *
+	 * @since 3.7
+	 *
+	 * @param array $post_data       A post array.
+	 * @param bool  $check_post_type Optional. Whether to check the post type in `$_GET`. Default is `true`.
+	 * @return array
+	 *
+	 * @phpstan-return array{}|array{from_post_id: int, new_lang: string}|never
+	 */
+	private function check_context( array $post_data, bool $check_post_type = true ): array {
+		if ( ! isset( $GLOBALS['pagenow'], $_GET['from_post'], $_GET['new_lang'], $post_data['post_type'] ) ) {
+			return array();
+		}
+
+		if ( $check_post_type && ( empty( $_GET['post_type'] ) || $_GET['post_type'] !== $post_data['post_type'] ) ) {
```

Comment:
Instead of having `$check_post_type`, what do you think to check `$post_data['post_type']` only when `$_GET['post_type']` is not empty?